includes/REST_API/Embed_Controller.php

```diff
@@ -222,30 +250,47 @@ private function get_data_from_post( $url ) {
 			$post_id = url_to_postid( $url );
 		}
 
-		if ( ! $post_id ) {
-			if ( $switched_blog ) {
-				restore_current_blog();
-			}
-
-			return false;
-		}
-
-		$post = get_post( $post_id );
-		if ( ! $post instanceof WP_Post ) {
-			return false;
+		if ( $post_id ) {
+			$post = get_post( $post_id );
 		}
 
-		if ( Story_Post_Type::POST_TYPE_SLUG !== $post->post_type ) {
-			return false;
+		if ( ! $post_id ) {
```

Comment:
```suggestion
		else {
```

Doesn't it make more sense to use an else here?

Reply:
The result is exactly the same, so just a matter of preference for coding style and readability.

Aside: PHPMD [by default](https://phpmd.org/rules/cleancode.html) complains about using `else` branches, but we've disabled that.